tsn/src/bin/throughput.rs

```diff
@@ -421,15 +420,16 @@ fn send_perf(sock: &i32, pkt: &mut Vec<u8>, size: usize) {
     }
 }
 
-fn make_send_pkt(ethernet_bytes: &Vec<u8>, pkt_info: &PktInfo) -> Vec<u8> {
-    let mut pkt: Vec<u8> = ethernet_bytes.clone();
-    println!("pkt = {:0x?}", pkt);
-    let mut pkt_info_bytes = bincode::serialize(pkt_info).unwrap();
-    println!("pkt_info_bytes = {:0x?}", pkt_info_bytes);
-    pkt.append(&mut pkt_info_bytes);
-    println!("pkt = {:0x?}", pkt);
-
-    pkt
+fn make_send_pkt(pkt: &mut Vec<u8>, ethernet: &Ethernet, pkt_info: &PktInfo, size: usize) {
+    let ethernet_bytes = bincode::serialize(&ethernet).unwrap();
+    let pkt_info_bytes = bincode::serialize(pkt_info).unwrap();
+    let ethernet_size = ethernet_bytes.len();
+    let pkt_info_size = pkt_info_bytes.len();
+
+    let new_len = ethernet_size + pkt_info_size;
+    pkt.splice(..ethernet_size, ethernet_bytes);
+    pkt.splice(ethernet_size..new_len, pkt_info_bytes);
+    pkt.truncate(size);
```

Comment:
Consider `split_at` method. it is cleaner way to do this.
https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_at

Reply:
I used `split_at_mut` instead if that's okay